### PR TITLE
BFD-1383: Update V2 line item allowed charge amount mapping

### PR DESCRIPTION
**Pull request by JeremyIR**
**JIRA Ticket:**
[BFD-1383](https://jira.cms.gov/browse/BFD-1383)

**User Story or Bug Summary:**

Use the allowed charge amount column instead of submitted charge amount in the V2 line item adjudication mapping.
---

### What Does This PR Do?

Modification of the transformerUtils in V2 to map the line item allowed charge amount value away from the submitted charge amount value
- changed V2 transformerUtils to map allowed charge amount and submitted charge amount values to their respective columns
- modified carrier and dme unit tests to account for updated mapping 
- updated endpoint response json files to include new line item allowed charge amount value

### What Should Reviewers Watch For?

Do the updated mapping in V2 transformerUtils and unit tests look correct and avoid any issues for end users if shipped?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by christrotteradhoc — CHANGES_REQUESTED** [review]
Looks good! I'll approve once we have the comms in place.

**Review by cbrunefearless — DISMISSED** [review]
LGTM

**Review by bfausett77 — DISMISSED** [review]
Looks great!  Nice job C-Brune!

**Review by bfausett77 — APPROVED** [review]

**Review by mackec1 — APPROVED** [review]
LGTM

**Review by mackec1 — DISMISSED** [review]

File: apps/bfd-server/dev/api-changelog.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
 # API Changelog
 
+## BFD-1383 Update V2 line item allowed charge amount mapping
```
Comment (ajsarkar128) [inline apps/bfd-server/dev/api-changelog.md:3]: I'd replace this title to something like "Submitted charge amount updated from submitted to allowed charge amount"

File: apps/bfd-server/dev/api-changelog.md [hunk 2]
```diff
@@ -1,5 +1,25 @@
...
+
+The charge amount for Carrier and DME have been updated from submitted to allowed.
```
Comment (ajsarkar128) [inline apps/bfd-server/dev/api-changelog.md:5]: Replace with "Previously, the allowed charge amount in the EOB FHIR response was incorrectly being populated by the submitted charge amount data field. The value for allowed charge amount is now being populated correctly by the respective allowed charge amount data field. Note that this is only for v2"

File: apps/bfd-server/dev/api-changelog.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 ## BFD-1383 Update V2 line item allowed charge amount mapping
 
-The charge amount for Carrier and DME have been updated from submitted to allowed.
+Previously, the allowed charge amount in the EOB FHIR response was incorrectly being populated by the submitted charge amount data field. The value for allowed charge amount is now being populated correctly by the respective allowed charge amount data field. Note that this is only for v2
```
Comment (bfausett77) [inline apps/bfd-server/dev/api-changelog.md:5]: Looks good!

**Review by bfausett77 — APPROVED** [review]
Looks good to me and the changelog has good descriptive information.

File: apps/bfd-server/bfd-server-war/src/test/resources/endpoint-responses/v2/eobByPatientIdAll.json [hunk 4]
```diff
@@ -5136,7 +5136,7 @@
             } ]
           },
           "amount" : {
-            "value" : 75.0,
+            "value" : 47.84,
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/test/resources/endpoint-responses/v2/eobByPatientIdAll.json:5139]: Assuming these were changed to be unique values; it's a good idea. We should ideally have all unique values for each input to ensure we don't have the wrong thing going in the wrong place (if not just verifying the input path value)

**Review by lsmitchell — APPROVED** [review]

**Comment by lsmitchell** [comment]
Also check ameet's comment about the title in the changlog.

**Pull request merged**